streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStore.java

```diff
@@ -92,11 +93,29 @@ Position getPosition() {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public <R> QueryResult<R> query(
         final Query<R> query,
         final PositionBound positionBound,
         final boolean collectExecutionInfo) {
 
+        if (query instanceof RangeQuery) {
+            final RangeQuery<Bytes, byte[]> typedQuery = (RangeQuery<Bytes, byte[]>) query;
+            final KeyValueIterator<Bytes, byte[]> keyValueIterator;
+            if (typedQuery.getLowerBound().isPresent() && typedQuery.getUpperBound().isPresent()) {
+                keyValueIterator = this.range(typedQuery.getLowerBound().get(), typedQuery.getUpperBound().get());
+            } else if (typedQuery.getLowerBound().isPresent()) {
+                keyValueIterator = this.range(typedQuery.getLowerBound().get(), null);
+            } else if (typedQuery.getUpperBound().isPresent()) {
+                keyValueIterator = this.range(null, typedQuery.getUpperBound().get());
+            } else {
+                keyValueIterator = this.range(null, null);
+            }
```

Comment:
It looks like this could be simplified. WDYT?

```suggestion
            keyValueIterator = this.range(typedQuery.getLowerBound().orElse(null), typedQuery.getUpperBound().orElse(null));
```

Reply:
Oh jeez /facepalm. Yes!!!!